### ISSUE TITLE
Nesting components with different `shadow` settings breaks in polyfilled browsers

### ISSUE THREAD

**Issue by mattdsteele** [ionic-team/stencil#447]
**Stencil version:**

```
 @stencil/core@0.2.3
```

**I'm submitting a:**

- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or https://stencil-worldwide.slack.com

**Current behavior:**

We have a number of nested components, some of which use `shadow: true`, others use `shadow: false` (mostly due to incomplete shadow dom polyfills). I've noticed that starting in `0.2.2`, these components behave in unexpected ways in polyfilled browsers.

Namely, I'm seeing this error in Firefox, IE11, and Edge:

```
TypeError: this.documentOwnerStyleInfo is undefined[Learn More]  mycomponent.core.pf.js:2049:7
	CustomStyle</CustomStyle.prototype.transformCustomStyleForDocument http://localhost:3333/build/mycomponent/mycomponent.core.pf.js:2049:7
	CustomStyle</CustomStyle.prototype.processStyles http://localhost:3333/build/mycomponent/mycomponent.core.pf.js:2041:11
	CustomStyle</CustomStyle.prototype.flushCustomStyles http://localhost:3333/build/mycomponent/mycomponent.core.pf.js:1978:26
	CustomStyle</CustomStyle.prototype.addStyle/</< http://localhost:3333/build/mycomponent/mycomponent.core.pf.js:2020:33
```

**Steps to reproduce:**

1. Clone https://github.com/mattdsteele/stencil-nested-shadows
2. `npm i`, `npm run dev --es5`
3. In Chrome styles are applied for the outer component (`shadow: false`) and the inner components (shadow set to `true` and `false`)
![image](https://user-images.githubusercontent.com/389077/35362862-b8129574-012c-11e8-9f43-1c680628a9e2.png)
4. In Firefox/Edge, styles are not consistently applied and there's a console error:
![image](https://user-images.githubusercontent.com/389077/35362887-dc713bd2-012c-11e8-9019-71861f0b4de2.png)

**Other information:**

In the demo repo, changing the parent to `shadow: true` causes styles to be applied, but the `TypeError` still remains. In our "real" app, the error persists and the styles aren't applied.

**Comment by jgw96** [comment]
Thanks for opening an issue with us! Note to self/team, this also happens with `scoped` styles.

**Comment by adamdbradley** [comment]
It appears this is fixed in master and what will become 0.4.0. Thanks for finding and testing this out!
